frontend_tests/node_tests/locking.js

```diff
@@ -0,0 +1,87 @@
+zrequire('locking');
+
+(function test_edge_cases() {
+    assert(!locking.is_topic_locked(undefined, undefined));
+    assert(!locking.is_topic_locked('nonexistent', undefined));
+}());
+
+(function test_basics() {
+    assert(!locking.is_topic_locked('devel', 'java'));
+    locking.add_locked_topic('devel', 'java');
+    assert(locking.is_topic_locked('devel', 'java'));
+
+    // test idempotentcy
+    locking.add_locked_topic('devel', 'java');
+    assert(locking.is_topic_locked('devel', 'java'));
+
+    locking.remove_locked_topic('devel', 'java');
+    assert(!locking.is_topic_locked('devel', 'java'));
+
+    // test idempotentcy
+    locking.remove_locked_topic('devel', 'java');
+    assert(!locking.is_topic_locked('devel', 'java'));
+
+    // test unknown stream is harmless too
+    locking.remove_locked_topic('unknown', 'java');
+    assert(!locking.is_topic_locked('unknown', 'java'));
+}());
+
+(function test_set_locked_topics() {
+    locking.set_locked_topics([
+        ['social', 'breakfast'],
+        ['design', 'typography'],
+    ]);
+    assert(locking.is_topic_locked('social', 'breakfast'));
+    assert(locking.is_topic_locked('design', 'typography'));
+    locking.set_locked_topics([
+        ['devel', 'java'],
+    ]);
+    assert(!locking.is_topic_locked('social', 'breakfast'));
+    assert(!locking.is_topic_locked('design', 'typography'));
+    assert(locking.is_topic_locked('devel', 'java'));
+}());
+
+(function test_case_insensitivity() {
+    locking.set_locked_topics([]);
+    assert(!locking.is_topic_locked('SOCial', 'breakfast'));
+    locking.set_locked_topics([
+        ['SOCial', 'breakfast'],
+    ]);
+    assert(locking.is_topic_locked('SOCial', 'breakfast'));
+    assert(locking.is_topic_locked('social', 'breakfast'));
```

Comment:
These tests are great!  As I mention elsewhere, we can just use stream ids to avoid some of the case-insensitivity issues here, but we'll still need tests for topic names.